### PR TITLE
Update Geo utils

### PR DESCRIPTION
**Pull request by theocalmes**

### CODE REVIEW

File: BotKit.xcodeproj/project.pbxproj.orig [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+				4DDE4A4116620785001707F4 /* Debug */,
+				4DDE4A4216620785001707F4 /* Release */,
+			);
+			defaultConfigurationIsVisible = 0;
+			defaultConfigurationName = Release;
+		};
+/* End XCConfigurationList section */
+	};
+	rootObject = 4DDE49FB16620785001707F4 /* Project object */;
+}
```
Comment (gfontenot) [inline BotKit.xcodeproj/project.pbxproj.orig:653]: I'm assuming you didn't mean to check in this file?

File: BotKit/Categories/UIView+Frame.h [hunk 2]
```diff
@@ -9,8 +9,12 @@
 CGPoint BKRectCenter(CGRect rect);
 CGRect BKCenterRect(CGRect rect, CGPoint center);
 CGRect BKScaleRect(CGRect rect, CGPoint scale);
+CGRect BKScaleRect1D(CGRect rect, float scale);
 
 CGPoint BKAddPoints(CGPoint p1, CGPoint p2);
+CGPoint BKSubPoints(CGPoint p1, CGPoint p2);
```
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.h:15]: Don't abbreviate Subtract here. Name it `BKSubtractPoints`.

File: BotKit/Categories/UIView+Frame.h [hunk 3]
```diff
@@ -9,8 +9,12 @@
...
+CGPoint BKScalePoint(CGPoint p1, CGPoint scale);
+CGPoint BKScalePoint1D(CGPoint p1, float scale);
```
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.h:17]: What does 1D mean in this context?

File: BotKit/Categories/UIView+Frame.m [hunk 4]
```diff
@@ -22,11 +22,40 @@ CGRect centerRect(CGRect rect, CGPoint center)
     return r;
 }
 
+CGRect BKScaleRect(CGRect rect, CGPoint scale)
+{
+    CGPoint iniCenter = BKRectCenter(rect);
+    CGRect scaled = CGRectMake(0, 0, rect.size.width*scale.x, rect.size.height*scale.y);
```
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.m:28]: You need spaces around the multiplication operators here.

File: BotKit/Categories/UIView+Frame.m [hunk 5]
```diff
@@ -22,11 +22,40 @@ CGRect centerRect(CGRect rect, CGPoint center)
     return r;
 }
 
+CGRect BKScaleRect(CGRect rect, CGPoint scale)
+{
+    CGPoint iniCenter = BKRectCenter(rect);
```
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.m:27]: I don't know what `iniCenter` means. Is there a better name for this variable?

File: BotKit/Categories/UIView+Frame.m [hunk 6]
```diff
@@ -22,11 +22,40 @@ CGRect centerRect(CGRect rect, CGPoint center)
...
+    CGRect scaled = CGRectMake(0, 0, rect.size.width*scale.x, rect.size.height*scale.y);
+    return BKCenterRect(scaled, iniCenter);
+}
+
+CGRect BKScaleRect1D(CGRect rect, float scale)
+{
+    CGPoint iniCenter = BKRectCenter(rect);
+    CGRect scaled = CGRectMake(0, 0, rect.size.width*scale, rect.size.height*scale);
```
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.m:35]: Same comments here as in the method above. Needs some whitespace, and a better name for `iniCenter`.

File: BotKit/Categories/UIView+Frame.h (hunk 3 above)
Comment (theocalmes) [inline BotKit/Categories/UIView+Frame.h:17]: 1D means 1 dimensionsal scale. ie we scale p1.x and p1.y by the same amount

File: BotKit/Categories/UIView+Frame.m (hunk 5 above)
Comment (theocalmes) [inline BotKit/Categories/UIView+Frame.m:27]: initialCenter?

File: BotKit/Categories/UIView+Frame.h (hunk 3 above)
Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.h:17]: So maybe `BKScalePointInSingleDimension`.

Comment (hyperspacemark) [inline BotKit/Categories/UIView+Frame.h:17]: There is precedent for using something like "1D". See `CLLocationCoordinate2D` and `CLLocationCoordinate2DMake()`.

Comment (theocalmes) [inline BotKit/Categories/UIView+Frame.h:17]: Well its not scaling in one dimension. Its that the scale is one dimentionsal.

Comment (gfontenot) [inline BotKit/Categories/UIView+Frame.h:17]: Fair enough.

**Pull request closed without merging**